storage/innobase/fsp/fsp0fsp.cc

```diff
@@ -4010,6 +3994,33 @@ dberr_t fil_space_t::garbage_collect(bool shutdown)
     return err;
   }
 
+  /* Reset the undo log segment slots in the rollback segment header
+  which exists in system tablespace. Undo cached segment will be
+  treated as unused file segment. These segment will be freed as a
+  part of inode_info::free_segs  */
```

Comment:
Grammar nitpick: "which exist" ("undo log segments lots" is plural) and "These segments".

In the commit message there is a similar grammatical error: "… InnoDB collects the segment from tables exist in system tablespace …"
There can’t be multiple predicates ("collects" and "exist") in the same sentence. We need a "that" or "which" in between: "tables that exist".